### PR TITLE
fs: do not call thread_rpc_free() when allocation fails

### PR DESCRIPTION
**Pull request by jforissier**
When thread_rpc_alloc_payload() fails, the cookie is not valid, so
thread_rpc_free() must not be called. This fixes a crash in xtest
7633 when shared memory is not large enough.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: core/arch/arm/tee/arch_tee_fs.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ int tee_fs_send_cmd(struct tee_fs_rpc *bf_cmd, void *data, uint32_t len,
 	thread_rpc_alloc_payload(sizeof(struct tee_fs_rpc) + len,
 				 &phpayload, &cpayload);
 	if (!phpayload)
-		goto exit;
+		return -1;
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/tee/arch_tee_fs.c]: Should `thread_rpc_alloc_payload()` and friends return error instead of relying on phpayload? May be in another PR?

**Comment by pascal-brand-st-dev** [comment]
With or without my comment:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jforissier** [comment]
Rebased with tags applied.
